app/src/util/cpk.ts

```diff
@@ -65,58 +59,15 @@ function standardizeTransaction(tx: Transaction): StandardTransaction {
 
 // Omen CPK monkey patch
 class OCPK extends CPK {
-  txCallbacks = new Map<RequestId, TxCallback>()
-
-  constructor(opts?: any) {
-    super(opts)
-
-    window.addEventListener('message', ({ data, origin }) => {
-      if (origin === window.origin) {
-        return
-      }
-
-      const { data: messagePayload, messageId, requestId } = data
-
-      const TRANSACTION_CONFIRMED = 'TRANSACTION_CONFIRMED'
-      const TRANSACTION_REJECTED = 'TRANSACTION_REJECTED'
-
-      if (messageId === TRANSACTION_CONFIRMED) {
-        const callback = this.txCallbacks.get(requestId)
-        if (callback) {
-          this.txCallbacks.delete(requestId)
-          callback.confirm({ safeTxHash: messagePayload.safeTxHash })
-        }
-      }
-      if (messageId === TRANSACTION_REJECTED) {
-        const callback = this.txCallbacks.get(requestId)
-        if (callback) {
-          this.txCallbacks.delete(requestId)
-          callback.reject(new Error('Transaction rejected'))
-        }
-      }
-    })
-  }
-
-  sendTransactions(transactions: StandardTransaction[], options?: ExecOptions): Promise<TransactionResult> {
-    const requestId = Math.trunc(Date.now())
-    return new Promise<TransactionResult>((confirm, reject) => {
-      const SEND_TRANSACTIONS_V2 = 'SEND_TRANSACTIONS_V2'
-      const txs = transactions.map(standardizeTransaction)
-      const params = { safeTxGas: options?.gas }
-      const data = {
-        txs,
-        params,
-      }
-      const message = {
-        messageId: SEND_TRANSACTIONS_V2,
-        requestId,
-        data,
-      }
-
-      this.txCallbacks.set(requestId, { confirm, reject })
-
-      window.parent.postMessage(message, '*')
-    })
+  async sendTransactions(transactions: StandardTransaction[], options?: ExecOptions): Promise<TransactionResult> {
```

Comment:
Is this necessary? Doesn't it have the same behaviour as the `cpk.execTransaction()` method when running as a Safe app inside the Multisig web interface?

Reply:
Yeah need this just to send safeTxGas with the newer version of safe-apps-sdk 👍